### ISSUE TITLE
[Haskell] ghc-mod: No such file or directory

### ISSUE THREAD

**Issue by Gustorn** [syl20bnr/spacemacs#1718]
After enabling `contrib/lang/haskell` it refuses to acknowledge that I have `ghc-mod` installed. `~/.cabal/bin` is added to `PATH`:
 ```bash
$ env | grep PATH
PATH=/home/gustorn/.cabal/bin:/home/gustorn/.local/bin: [rest of PATH]
```

My `.spacemacs` file (.el extension added for syntax highlighting): [.spacemacs](https://gist.github.com/Gustorn/ded7144a9534886be270) (I already tried with `flyspell` not excluded)

The actual error:
```
Initializing...
"ghc-mod" not found
File mode specification error: (file-error "Searching for program" "no such file or directory" "ghc-mod")
Spacemacs is up to date.
```

`:company-ghc-diagnose`
```
* company-ghc backend found: company-ghc
* automatic scan module is enabled
* ghc-boot process has not been done

Module                                  Alias               Candidates
-------------------------------------------------------------------------------
Prelude   
```

Using `:!ghc-mod` inside emacs finds the executable just fine.

OS: Linux
Emacs: GNU Emacs 24.5.1

**Comment by codygman** [comment]
What does `which ghc-mod` give you inside of an eshell?

**Comment by Gustorn** [comment]
It doesn't find it, so I'm guessing this line: `(add-to-list 'exec-path "~/.cabal/bin/")` is either at the wrong place or is just plain wrong. I'm fairly new to emacs, so I apologize if the solution is really obvious for more experienced people.

**Comment by syl20bnr** [comment]
You can inspect the value of the variable with `SPC h d  exec-path RET`.

**Comment by tuhdo** [comment]
It should be `SPC h d v exec-path RET`.

**Comment by Gustorn** [comment]
Alright, I solved the problem: `exec-path` was not picked up from the `dotspacemacs/init` function; adding the same line to `dotspacemacs/config` solved the problem, which means either something isn't quite working right in spacemacs, or if this is the intended behaviour, then the Haskell layer readme should be updated (or it's completely possible that the error is on my end, but I only changed some of the layers that are loaded and the default themes, otherwise this was a completely fresh install).

**Comment by syl20bnr** [comment]
`dotspacemacs/init` is exectued before the initialization of the layers. Possibly somewhere in the layers `exec-path` is overwritten (could be also in a package but it less likely).
`dotspacemacs/config` is executed after the initialization of the layers.

**Comment by Gustorn** [comment]
I get that, but here's a quote from the Haskell layer readme:

"or [add this path] to the Emacs `exec-path` variable in the `dotspacemacs/init` function of your `.spacemacs` file:" This doesn't work, while adding it to `dotspacemacs/config` works just fine.

**Comment by luhkevin** [comment]
Gustorn -- thank you for this fix! But not for the Haskell layer. I was setting up gtgas as per the instructions here: https://github.com/syl20bnr/spacemacs/tree/master/contrib/gtags

When I tried to use helm-gtags, or invoke any of the installed programs such as gtags or global, spacemacs would say that there was "no such file or directory"

So I added "/usr/bin" -- where global is installed -- to the "exec-path" and helm-gtags worked. 

I looked at the path with `SPC h d v exec-path RET.`, and it is definitely not the same as the PATH variable in my shell....in fact it only includes: 
("/usr/bin" "%s/" "/usr/lib/emacs/24.5/x86_64-unknown-linux-gnu/")

Is this supposed to happen? Or is spacemacs doing something funny?
